File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
from collections import deque

def BFS(V):
    q = deque([V]) 
    visited2[V] = True  # 첫 시작점은 방문 True로 초기화
    while q:  
        V = q.popleft()  
        print(V, end=" ") 
        for i in range(1, N + 1):  
            if not visited2[i] and graph[V][i]:  # 방문 했던 곳이 아니고, V 값과 연결되어 있는지
                q.append(i) 
                visited2[i] = True  # 방문(True)으로 초기화


def DFS(V):
    visited1[V] = True 
    print(V, end=" ")
    for i in range(1, N + 1):
        if not visited1[i] and graph[V][i]:  # 방문 했던 곳이 아니고, V 값과 연결되어 있는지
            DFS(i)  # 재귀 호출을 이용하여 깊은 탐색

N, M, V = map(int, input().split())

graph = [[False] * (N + 1) for _ in range(N + 1)]

for _ in range(M):
    a, b = map(int, input().split())
    graph[a][b] = True
    graph[b][a] = True

visited1 = [False] * (N + 1)  # DFS 방문 유무
visited2 = [False] * (N + 1)  # BFS 방문 유무

DFS(V)
print()
BFS(V)

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.ArrayDeque;
import java.util.Deque;
import java.util.StringTokenizer;


public class Main {
	static boolean[] visited1;
	static boolean[] visited2;
	static boolean[][] graph;
	static int N;
	static int M; 
	static int V;
	

    public static void main(String[] args) throws Exception {
    	BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    	StringTokenizer st = new StringTokenizer(br.readLine());
    	N = Integer.parseInt(st.nextToken());
    	M = Integer.parseInt(st.nextToken());
    	V = Integer.parseInt(st.nextToken());
    	
    	graph = new boolean[N+1][N+1];
    	
    	for(int i = 0 ; i < M ; i++) {
    		st = new StringTokenizer(br.readLine());
    		int a = Integer.parseInt(st.nextToken());
    		int b = Integer.parseInt(st.nextToken());
    		graph[a][b] = true;
    		graph[b][a] = true;
    	}
    	
    	visited1 = new boolean[N+1];
    	visited2 = new boolean[N+1];
    	
    	DFS(V);
    	System.out.println();
    	BFS();
    }

	private static void BFS() {
		// TODO Auto-generated method stub
		Deque<Integer> q =  new ArrayDeque<Integer>();
		q.offer(V);
		visited2[V] = true;
		while(!q.isEmpty()){
			V = q.poll();
			System.out.print(V + " ");
			for(int i = 1 ; i < N+1 ; i++) {
				if(!visited2[i] && graph[V][i]) {
					q.offer(i);
					visited2[i] = true;
				}
			}
		}
		
	}

	private static void DFS(int a) {
		// TODO Auto-generated method stub
		visited1[a] = true;
		System.out.print(a + " ");
		for(int i = 1 ; i < N+1 ; i++) {
			
			if(!visited1[i] && graph[a][i]) {
				DFS(i);
			}
		}
		
	}
}
